### PR TITLE
[TS] LPS-159513 ArrayStoreException occurs on large number in repeatable Numeric field

### PR DESCRIPTION
**Pull request by IstvanD**
Dear Team,

Could you please review this pull request?

If a numeric field value exceeds `Integer.MAX_VALUE`, it will end up in the `List<Serializable> values` as a String.
https://github.com/liferay/liferay-portal/blob/master/modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/storage/constants/FieldConstants.java#L121-L124

Therefore, before converting the `List` to `Array[Number]`, all values must be converted to Number to avoid `ArrayStoreException` due to `String => Number` conversion.

Thank you and best regards,
István

### CODE REVIEW

**Comment by rodrigopaulino** [comment]
ci:test:sf

**Comment by rodrigopaulino** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/storage/constants/FieldConstants.java [hunk 1]
```diff
@@ -187,15 +188,20 @@ public static Serializable getSerializable(
...
+		else if (type.equals(FieldConstants.DOUBLE) ||
+				 type.equals(FieldConstants.INTEGER)) {
+
+			List<Number> numberValues = new ArrayList<>();
+
+			for (Serializable value : values) {
+				numberValues.add(GetterUtil.getNumber(value));
+			}
+
+			return numberValues.toArray(new Number[0]);
```
Comment (rodrigopaulino) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/storage/constants/FieldConstants.java:200]: This logic already exists in Liferay. Check this out:
```suggestion
			return ListUtil.toArray(
				values,
				new Accessor<Object, Number>() {

					@Override
					public Number get(Object value) {
						return GetterUtil.getNumber(value);
					}

					@Override
					public Class<Number> getAttributeClass() {
						return Number.class;
					}

					@Override
					public Class<Object> getTypeClass() {
						return Object.class;
					}

				});
```

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/storage/constants/FieldConstants.java [hunk 2]
```diff
@@ -187,15 +188,20 @@ public static Serializable getSerializable(
 		else if (type.equals(FieldConstants.DATE)) {
 			return values.toArray(new String[0]);
 		}
-		else if (type.equals(FieldConstants.DOUBLE)) {
-			return values.toArray(new Number[0]);
+		else if (type.equals(FieldConstants.DOUBLE) ||
+				 type.equals(FieldConstants.INTEGER)) {
```
Comment (rodrigopaulino) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/storage/constants/FieldConstants.java:192]: Why did you add integer here as well?

**Review by rodrigopaulino — CHANGES_REQUESTED** [review]
Hi, @IstvanD.
Thanks for you PR. 
I just have one question and one suggestion that you can find below. 🙂

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/storage/constants/FieldConstants.java (hunk 1 above)
Comment (IstvanD) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/storage/constants/FieldConstants.java]: hi @rodrigopaulino thanks for the suggestion! I have applied and force-pushed it to this pull.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/storage/constants/FieldConstants.java (hunk 2 above)
Comment (rodrigopaulino) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/storage/constants/FieldConstants.java:193]: Got it. Thanks for explaining.

**Review by rodrigopaulino — APPROVED** [review]
LGTM!

**Comment by rodrigopaulino** [comment]
ci:forward
